iceoryx_dust/vocabulary/include/iox/span.hpp

```diff
@@ -0,0 +1,451 @@
+// Copyright (c) 2022 - 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_VOCABULARY_SPAN_HPP
+#define IOX_DUST_VOCABULARY_SPAN_HPP
+
+#include "iceoryx_hoofs/cxx/requires.hpp"
+#include "iox/detail/span_iterator.hpp"
+#include "iox/detail/uninitialized_array_type_traits.hpp"
+#include "iox/type_traits.hpp"
+#include "iox/uninitialized_array.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <cstdio>
+#include <limits>
+
+using namespace std;
+
+namespace iox
+{
+// constants
+constexpr uint64_t DYNAMIC_EXTENT = std::numeric_limits<uint64_t>::max();
+
+// Implementation of C++17 std::size() and std::data()
+
+/// @brief Returns c.size(), converted to the return type if necessary.
+/// @tparam C Type of the container
+/// @param c A container or view with a size member function
+/// @return The size of c
+template <typename Container>
+constexpr auto size(const Container& container) -> decltype(container.size());
```

Comment:
I'm a bit torn here since the overload might have a different return type than the function below or in `size.hpp`

Reply:
@elBoberido That's how it's defined [in the STL](https://en.cppreference.com/w/cpp/iterator/size). Therefore I would leave it as it is.